code/parachain/frame/composable-traits/src/assets.rs

```diff
@@ -34,3 +35,105 @@ pub struct Asset<Balance, ForeignId> {
 	pub foreign_id: Option<ForeignId>,
 	pub existential_deposit: Balance,
 }
+
+pub trait AssetTypeInspect {
+	type AssetId;
+
+	fn inspect(asset: &Self::AssetId) -> AssetType;
+}
+
+pub enum AssetType {
+	Foreign,
+	Local,
+}
+/// Routing of indepent parts of the `AssetMetadata` from `pallet-assets-registry`
+pub trait InspectRegistryMetadata {
+	type AssetId;
+
+	/// Return the name of an asset.
+	fn asset_name(asset_id: &Self::AssetId) -> Option<Vec<u8>>;
+	/// Return the symbol of an asset.
+	fn symbol(asset_id: &Self::AssetId) -> Option<Vec<u8>>;
+	/// Return the decimals of an asset.
+	fn decimals(asset_id: &Self::AssetId) -> Option<u8>;
+}
+
+pub trait MutateRegistryMetadata {
+	type AssetId;
+	type BoundedName;
+	type BoundedSymbol;
+
+	/// Sets the metadata of an asset
+	fn set_metadata(
+		asset_id: &Self::AssetId,
+		name: Vec<u8>,
+		symbol: Vec<u8>,
+		decimals: u8,
+	) -> DispatchResult;
+
+	/// Update the metadata of an asset.
+	///
+	/// All metadata feilds are optional, only those provided as `Some` will be updated, the
+	/// rest will be unchanged.
+	fn update_metadata(
+		asset_id: &Self::AssetId,
+		name: Option<Vec<u8>>,
+		symbol: Option<Vec<u8>>,
+		decimals: Option<u8>,
+	) -> DispatchResult;
+}
+
+/// Structure to represent basic asset metadata such as: name, symbol, decimals.
+#[derive(Clone, Encode, Decode, Eq, PartialEq, Default, RuntimeDebug, MaxEncodedLen, TypeInfo)]
+pub struct AssetMetadata<BoundedName, BoundedSymbol> {
+	/// Name of the asset.
+	pub name: BoundedName,
+	/// Symbol of the asset.
+	pub symbol: BoundedSymbol,
```

Comment:
These could just be bounded vecs directly with the generic parameter being a `Get` type

Reply:
So, `(AssetId) -> Vec<???>`

Reply:
`BoundedVec<u8, MaxNameLen>` where `MaxNameLen: Get<u32>`